bebop/joystick.cc

```diff
@@ -1,73 +1,66 @@
-#include "joystick.h"
+// C++ includes
 #include <stdexcept>
 
+// local includes
+#include "joystick.h"
+
 namespace GeNNRobotics {
 namespace Robots {
 BebopJoystick::BebopJoystick(Bebop *bebop)
   : m_Bebop(bebop)
 {
-    if (!m_Joystick.open()) {
-        throw std::runtime_error("Could not find joystick");
-    }
-    m_Joystick.startThread(EventCallback, this);
+    setCallback([=](HID::Event &js) { eventCallback(js); });
```

Comment:
I think it's nice to be more specific in your capture expression i.e. the [] bit - in this case you only need to capture ``[this]``

Reply:
Agreed